quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusProjectInfoCache.java

```diff
@@ -41,18 +46,27 @@ public QuarkusProjectInfoCache(QuarkusProjectInfoProvider provider) {
 	 */
 	public CompletableFuture<QuarkusProjectInfo> getQuarkusProjectInfo(QuarkusProjectInfoParams params) {
 		// Search project info in cache
-		QuarkusProjectInfo projectInfo = super.get(params.getUri());
+		QuarkusProjectInfo projectInfo = cache.get(params.getUri());
 		if (projectInfo == null) {
 			// not found in cache, load the project info from the JDT LS Extension
 			return provider.getQuarkusProjectInfo(params).thenApplyAsync(info ->
 			// information was loaded, update the cache
 			{
-				QuarkusProjectInfoCache.this.put(params.getUri(), info);
+				cache.put(params.getUri(), info);
 				return info;
 			});
 		}
 		// Returns the cached project info
 		return CompletableFuture.completedFuture(projectInfo);
 	}
 
+	public void resetCache(Set<String> projects) {
+		List<String> uris = cache.entrySet().stream()
+				.filter(entry -> projects.contains(entry.getValue().getProjectName())).map(Map.Entry::getKey)
+				.collect(Collectors.toList());

```

Comment:
call cache.remove in .each(), instead of creating a new list os strings

Reply:
Mutating the streamed collection might not work at all ;-)

Reply:
I want to have list of uris to execute in the future validation of each uris (application.properties)